lib/internal/process/esm_loader.js

```diff
@@ -1,21 +1,59 @@
 'use strict';
 
+const {
+  ArrayPrototypePush,
+  PromisePrototypeThen,
+  ReflectApply,
+  SafeMap,
+} = primordials;
+
+const assert = require('internal/assert');
 const { createModuleLoader } = require('internal/modules/esm/loader');
 const { getOptionValue } = require('internal/options');
 const {
   hasUncaughtExceptionCaptureCallback,
 } = require('internal/process/execution');
 const { pathToFileURL } = require('internal/url');
-const { kEmptyObject } = require('internal/util');
+const { kEmptyObject, createDeferredPromise } = require('internal/util');
 
 let esmLoader;
+let init = false;
 
 module.exports = {
   get esmLoader() {
-    return esmLoader ??= createModuleLoader(true);
+    return esmLoader ??= { __proto__: null, cjsCache: new SafeMap(), import() {
+      const { promise, resolve, reject } = createDeferredPromise();
+      ArrayPrototypePush(this.importRequests, { arguments: arguments, resolve, reject });
+      return promise;
+    }, importRequests: [] };
+  },
+  initIfNeeded() {
+    // TODO: we could try to avoid loading ESM loader on CJS-only codebase
```

Comment:
I don't think we should initialize the ESM loader in CJS-only code base. At the very least the ESM internals must be snapshotable before we consider this.